apps/examples/testcase/ta_tc/mqtt/utc/utc_mqtt_main.c

```diff
@@ -0,0 +1,479 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <sys/socket.h>
+#include <errno.h>
+#include <apps/shell/tash.h>
+#include <mqtt/mqtt_api.h>
+#include "tc_common.h"
+
+#define UTC_MQTT_TOPIC "topic_utc"
+#define UTC_PUB_CLIENT_ID "utc_mqtt_pub/02"
+#define UTC_SUB_CLIENT_ID "utc_mqtt_sub/01"
+#define UTC_MQTT_MESSAGE "utc hello mqtt"
+#define UTC_MQTT_LOGE printf("%s:%d\n", __FUNCTION__, __LINE__)
+#define UTC_MQTT_DEBUG
+#ifdef UTC_MQTT_DEBUG
+#define UTC_MQTT_LOGD printf
+#else
+#define UTC_MQTT_LOGD
+#endif
+#define UTC_MQTT_WAIT_SIGNAL \
+	do {										\
+		pthread_mutex_lock(&g_ocfmutex);			\
+		printf("[pksignal] T%d wait signal\n", getpid());\
+		pthread_cond_wait(&g_ocfcond, &g_ocfmutex);	\
+		pthread_mutex_unlock(&g_ocfmutex);			\
+	} while (0)
+		
+#define UTC_MQTT_SEND_SIGNAL \
+	do {										\
+		pthread_mutex_lock(&g_ocfmutex);			\
+		printf("[pksignal] T%d send signal\n", getpid());\
+		pthread_cond_signal(&g_ocfcond);			\
+		pthread_mutex_unlock(&g_ocfmutex);			\
+	} while (0)	
+
+
+static char g_mqtt_msg[] = UTC_MQTT_MESSAGE;
+static pthread_mutex_t g_ocfmutex = PTHREAD_MUTEX_INITIALIZER;;
+static pthread_cond_t g_ocfcond;	
+static mqtt_client_t *g_mqtt_pub_handle = NULL;
+static mqtt_client_t *g_mqtt_sub_handle = NULL;
+
+/**
+ * Private Functions
+ */
+int _utc_mqtt_init(void)
```

Comment:
This function is using only in this file. Let's add *static*.

Reply:
done